slither/analyses/data_dependency/data_dependency.py

```diff
@@ -161,11 +162,27 @@ def pprint_dependency(context):
 ###################################################################################
 ###################################################################################
 
+return_values = {}
+
 def compute_dependency(slither):
 
     slither.context[KEY_INPUT] = set()
     slither.context[KEY_INPUT_SSA] = set()
 
+    # Generate a mapping of functions to their return values
+    for contract in slither.contracts:
+        for function in contract.all_functions_called:
+            found_return = False
+            if (not found_return):
+                for node in function.nodes:
+                    if (node.type == NodeType.RETURN):
+                        found_return = True
+                        for ir in node.irs_ssa:
+                            if isinstance(ir, Return):
+                                return_values[function] = ir.values
```

Comment:
We should use a `collections.defaultdict` and the operator `+=`, otherwise function with multiple return statement will not work

Reply:
Sorry, my bad. Multiple returns is an obvious bug-will fix.

Will also look into memoization of return values.